components/renku_data_services/data_connectors/db.py

```diff
@@ -262,12 +262,17 @@ async def _insert_data_connector(
             )
             project_slug = await session.scalar(
                 select(ns_schemas.EntitySlugORM)
+                .where(ns_schemas.EntitySlugORM.namespace_id == ns.id)
```

Comment:
If this line is removed, then the condition below can trigger and send back "Mismatched project slug 'user-1/project-a` and data connector namespace 'user-2/project-1'".